app/javascript/dashboard/helper/URLHelper.js

```diff
@@ -64,3 +64,37 @@ export const isValidURL = value => {
     /^https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)$/gm;
   return URL_REGEX.test(value);
 };
+
+export const getArticleSearchURL = ({
+  host,
+  portalSlug,
+  pageNumber,
+  locale,
+  status,
+  authorId,
+  categorySlug,
+  sort,
+  query,
+}) => {
+  const queryParams = new URLSearchParams({});
+
+  // Define the parameters in an array
+  const params = [
+    { key: 'page', value: pageNumber },
+    { key: 'locale', value: locale },
+    { key: 'status', value: status },
+    { key: 'author_id', value: authorId },
+    { key: 'category_slug', value: categorySlug },
+    { key: 'sort', value: sort },
+    { key: 'query', value: query },
+  ];
+
+  // Add parameters to queryParams
+  params.forEach(({ key, value }) => {
+    if (value) {
```

Comment:
This will parse `0` as false and `[]` as true, is this okay to have?

Reply:
Makes sense, I have added specific check for undefined and null in the condition.